packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ListView/DynamicTable/DeleteButton/index.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import DeleteIcon from '@strapi/icons/DeleteIcon';
+import { IconButton } from '@strapi/parts/IconButton';
+import { Box } from '@strapi/parts/Box';
+import { stopPropagation } from '@strapi/helper-plugin';
+import { useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+
+const DeleteButton = ({ canDelete, tokenName, onClick }) => {
```

Comment:
Instead of adding additional props to this DeleteButton, WYT of conditionally render this one from the outside?

- It reduces the number of props passed
- It clearly shows in the parent that you only show it when you are allowed to


```jsx
// in the parent

{canDelete && <DeleteButton onClick={} tokenName={tokenName} />}
```